native/machine-coding/sudoku/script.js

```diff
@@ -0,0 +1,410 @@
+function possible(board, y, x, n) {
+  for (let i = 0; i < 9; i++) {
+    if (board[y][i] === n || board[i][x] === n) {
+      return false;
+    }
+  }
+
+  const xSquare = Math.floor(x / 3) * 3;
+  const ySquare = Math.floor(y / 3) * 3;
+
+  for (let i = 0; i < 3; i++) {
+    for (let j = 0; j < 3; j++) {
+      if (board[ySquare + i][xSquare + j] === n) {
+        return false;
+      }
+    }
+  }
+
+  return true;
+}
+
+function solveSudoku(board) {
+  const solutions = [];
+
+  function backtrack(board, y, x) {
+    if (y === 9) {
+      solutions.push(JSON.parse(JSON.stringify(board)));
+      return;
+    }
+
+    if (board[y][x] !== 0) {
+      if (x === 8) {
+        backtrack(board, y + 1, 0);
+      } else {
+        backtrack(board, y, x + 1);
+      }
+      return;
+    }
+
+    for (let n = 1; n <= 9; n++) {
+      if (possible(board, y, x, n)) {
+        board[y][x] = n;
+
+        if (x === 8) {
+          backtrack(board, y + 1, 0);
+        } else {
+          backtrack(board, y, x + 1);
+        }
+
+        board[y][x] = 0;
+      }
+    }
+  }
+
+  backtrack(board, 0, 0);
+
+  return solutions;
+}
+
+function shuffle(arr) {
+  for (let i = arr.length - 1; i > 0; i--) {
+    const j = Math.floor(Math.random() * (i + 1));
+    [arr[i], arr[j]] = [arr[j], arr[i]];
+  }
+}
+
+function isValid(board, row, col, num) {
+  for (let i = 0; i < 9; i++) {
+    if (board[row][i] === num || board[i][col] === num) {
+      return false;
+    }
+  }
+
+  const boxRow = Math.floor(row / 3) * 3;
+  const boxCol = Math.floor(col / 3) * 3;
+  for (let i = boxRow; i < boxRow + 3; i++) {
+    for (let j = boxCol; j < boxCol + 3; j++) {
+      if (board[i][j] === num) {
+        return false;
+      }
+    }
+  }
+
+  return true;
+}
+
+function generateSudoku() {
+  const board = Array(9)
+    .fill()
+    .map(() => Array(9).fill(0));
+
+  const nums = [1, 2, 3, 4, 5, 6, 7, 8, 9];
+  shuffle(nums);
+  for (let i = 0; i < 9; i++) {
+    board[0][i] = nums[i];
+  }
+
+  function fill(row, col) {
+    if (col === 9) {
```

Comment:
dont hard code numbers like 9.
Use size as variable on global and derive everything from that size